src/libmongoc/tests/test-mongoc-client.c

```diff
@@ -675,24 +675,6 @@ test_mongoc_client_authenticate_cached (bool pooled)
       }
    }
 
-   /* screw up the cache */
```

Comment:
With this change, the cache can no longer be tampered with as such. Tampering with the local cache gets fixed during the next call to `_mongoc_cluster_auth_scram_continue`, specifically at this line:
```C
   cluster->scram_cache = _mongoc_scram_get_cache (scram);
```